app/src/main/java/com/minimalisticapps/priceconverter/presentation/home/viewmodels/HomeViewModel.kt

```diff
@@ -136,32 +131,114 @@ class HomeViewModel @Inject constructor(
     //    function for getting fiat coins
     fun getFiatCoins() {
         viewModelScope.launch {
-            getFiatCoinUseCase().collect {
-                var position = -1
-                _fiatCoinsList.value = it.map { bitPayCoinWithFiatCoin ->
-                    Pair(position++, bitPayCoinWithFiatCoin)
+            getFiatCoinUseCase().collect { it ->
+                val listStateValue: MutableList<Pair<Int, BitPayCoinWithFiatCoin>> = mutableListOf()
+                val shitcoinInputsStateValue: MutableMap<String, MutableState<TextFieldValue>> =
+                    mutableMapOf()
+
+                it.forEachIndexed { index, shitcoin ->
+                    listStateValue.add(Pair(index, shitcoin))
+
+                    val code = shitcoin.fiatCoinExchange.code
+                    shitcoinInputsStateValue[code] = shitcoinInputsState[code]
+                        ?: mutableStateOf(TextFieldValue())
+                }
+
+                _shitcoinListState.value = listStateValue
+                shitcoinInputsState = shitcoinInputsStateValue
+
+                if (selectedCoin.value == "BTC") {
+                    recalculateByBitcoin()
+                } else {
+                    val index =
+                        _shitcoinListState.value.find { it2 -> it2.second.fiatCoinExchange.code == selectedCoin.value }?.first
```

Comment:
On reload we need to recalculate the amounts in relation to a selected shitcoin or BTC, For this the `selectedCoin` was added to the model